src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -63,110 +66,56 @@ private[spark] object Config {
           "the spark driver and its executors are deployed to the same cluster."
       )
       .stringConf
-      .checkValue(selectorsValidator, "Selectors must be valid kubernetes labels/selectors")
-      .createWithDefaultString(DEFAULT_CLUSTER_SELECTORS)
-
-  private[armada] val singleLabelOrNone: Regex = s"^($label)?$$".r
-
-  private[armada] def singleLabelValidator(l: CharSequence): Boolean = {
-    singleLabelOrNone.findPrefixMatchOf(l).isDefined
-  }
+      .checkValue(k8sLabelListValidator, invalidLabelListErrorMessage)
+      .createWithDefaultString("")
 
-  val GANG_SCHEDULING_NODE_UNIFORMITY_LABEL: ConfigEntry[String] =
-    ConfigBuilder("spark.armada.scheduling.nodeUniformityLabel")
-      .doc("A single kubernetes label to apply to both driver and executors")
+  val ARMADA_JOB_GANG_SCHEDULING_NODE_UNIFORMITY: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.scheduling.nodeUniformity")
+      .doc(
+        "Constrains the jobs that make up a gang to be scheduled across a uniform set of nodes. " +
+          "Specifically, if set, all gang jobs are scheduled onto nodes for which the value of the provided label is equal."
+      )
       .stringConf
-      .checkValue(singleLabelValidator, "Label must be a valid kubernetes label, just the label!")
+      .checkValue(K8sValidator.Label.isValidKey, "Only a valid RFC 1123 label key is allowed!")
       .createWithDefaultString("armada-spark")
 
-  private val validServiceNamePrefix: Regex = "([a-z][0-9a-z-]{0,29})?".r
-
-  private[armada] def serviceNamePrefixValidator(name: CharSequence): Boolean = {
-    validServiceNamePrefix.findPrefixMatchOf(name).isDefined
-  }
-
-  val DRIVER_SERVICE_NAME_PREFIX: ConfigEntry[String] =
-    ConfigBuilder("spark.armada.driverServiceNamePrefix")
+  val SPARK_DRIVER_SERVICE_NAME_PREFIX: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.driver.serviceNamePrefix")
       .doc(
         "Defines the driver's service name prefix within Armada. Lowercase a-z and '-' characters only. " +
           "Max length of 30 characters."
       )
       .stringConf
       .checkValue(
-        serviceNamePrefixValidator,
-        "Service name prefix must adhere to rfc 1035 label names."
+        K8sValidator.Name.isValidPrefix,
+        "Service name prefix must adhere to RFC 1035 label names."
       )
       .createWithDefaultString("armada-spark-driver-")
 
-  private val invalidLabelListErrorMessage =
-    "Must be a comma-separated list of valid Kubernetes labels (each as key=value)"
-
-  val ARMADA_SPARK_GLOBAL_LABELS: ConfigEntry[String] =
-    ConfigBuilder("spark.armada.global.labels")
+  val ARMADA_SPARK_POD_LABELS: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.pod.labels")
       .doc(
-        "A comma separated list of kubernetes labels (in key=value format) to be added to all " +
+        "A comma-separated list of kubernetes labels (in key=value format) to be added to all " +
           "both the driver and executor pods."
```

Comment:
I think `to all both the driver and executor pods` sounds strange, `to both the driver and executor pods` is sufficient:
```suggestion
        "A comma-separated list of kubernetes labels (in key=value format) to be added to  " +
          "both the driver and executor pods."
```

Reply:
yep, typo